File: array/1.rb
#coding: utf-8
'''
Даны два массива целых чисел одинаковой длины A[0..n−1] и B[0..n−1]
Необходимо найти первую пару индексов i0 и j0, i0≤j0, такую что A[i0]+B[j0]=max{A[i]+B[j],где0≤i<n,0≤j<n,i≤j}.
Время работы – O(n).
Ограничения: 1≤n≤100000,0≤A[i]≤100000,0≤B[i]≤1000001≤n≤100000,0≤A[i]≤100000,0≤B[i]≤100000  для любого i.
Sample Input:
4
4 -8 6 0
-10 3 1 1
Sample Output:
0 1
'''
n = gets
a = gets.split(' ').map(&:to_i)
b = gets.split(' ').map(&:to_i)

i0, j0 = 0, 0
max = a[0]+b[0]

a[1..-1].each_with_index{|e_a, i|
	b[0..i].each{ |e_b| max = e_a+e_b if e_a+e_b>max }
}
puts max

File: gcd/1.py
'''
НОД

По данным двум числам 1≤a,b≤2⋅10^9 1≤a,b≤2⋅10^9 найдите их наибольший общий делитель.

Sample Input 1:
18 35
Sample Output 1:
1

Sample Input 2:
14159572 63967072
Sample Output 2:
4
'''

def gcd(a,b):
  if a==0:
    return b
  if b==0:
    return a
  if a<b:
    return gcd(a, b%a)
  return gcd(a%b, b)

a,b = [int(i) for i in input().split()]

print(gcd(a,b))
